includes/Story_Renderer.php

```diff
@@ -213,44 +286,39 @@ protected function replace_body_end_tag( $content ) {
 
 		$output = (string) ob_get_clean();
 
-		return str_replace( '</body>', $output . '</body>', $content );
-	}
+		if ( empty( $output ) ) {
+			return;
+		}
 
-	/**
-	 * Renders the story.
-	 *
-	 * @return string The complete HTML markup for the story.
-	 */
-	public function render() {
-		$markup = $this->post->post_content;
-		$markup = $this->replace_html_start_tag( $markup );
-		// Add before replace_html_head to leverage the `web_stories_story_head` action.
-		$markup = $this->maybe_add_analytics( $markup );
-		$markup = $this->replace_html_head( $markup );
-		$markup = $this->add_poster_images( $markup );
-		$markup = $this->add_publisher_logo( $markup );
-		$markup = $this->replace_body_start_tag( $markup );
-		$markup = $this->replace_body_end_tag( $markup );
-		return $markup;
+		$new_content = $this->string_to_doc( $output );
+
+		if ( ! $new_content->documentElement ) {
+			return;
+		}
+
+		/* @var DOMElement $body The <body> element. */
+		$body = $this->document->getElementsByTagName( 'body' )[0];
```

Comment:
A line that looks like this is repeated 7 times. How about a protected method. 
Something like 
```
protected function get_tag_by_name( $tag_name ){
      return $this->document->getElementsByTagName( $tag_name )[0];
}
```